src/prefect/tasks/mysql/mysql.py

```diff
@@ -173,13 +178,27 @@ def run(
                 "The 'fetch' parameter must be one of the following - ('one', 'many', 'all')"
             )
 
+        cursor_types = {
+            "cursor": pymysql.cursors.Cursor,
+            "dictcursor": pymysql.cursors.DictCursor
+        }
+
+        if callable(cursor_type):
+            cursor_class = cursor_type
+        else:
+            cursor_class = cursor_types.get(cursor_type.lower())
```

Comment:
Note that was written, this does not cover the case where the cursor is not callable and not a string (which was something Jim alluded to) so if someone passed a list in it'd fail on this `.lower()` call and be a confusing error.

```suggestion
        elif isinstance(cursor_type, str):
            cursor_class = cursor_types.get(cursor_type.lower())
        else:
            cursor_class = None  # will be raised below
```